pkg/kubectl/cmd/expose.go

```diff
@@ -133,7 +134,11 @@ func RunExpose(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
 	}
 	names := generator.ParamNames()
 	params := kubectl.MakeParams(cmd, names)
-	params["default-name"] = info.Name
+	name := info.Name
+	if len(name) > validation.DNS952LabelMaxLength {
+		name = name[:validation.DNS952LabelMaxLength]
```

Comment:
I think the logic is already there, since we pass --name as a separate
parameter so validation will take care of it once we try to create. I'll
add a separate test case.

On Fri, Oct 2, 2015 at 9:32 AM, Clayton Coleman <notifications@github.com>
wrote:

> In pkg/kubectl/cmd/expose.go
> <https://github.com/kubernetes/kubernetes/pull/14728#discussion_r40997045>
> :
>
> > @@ -133,7 +134,11 @@ func RunExpose(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
> >  	}
> >  	names := generator.ParamNames()
> >  	params := kubectl.MakeParams(cmd, names)
> > -	params["default-name"] = info.Name
> > +	name := info.Name
> > +	if len(name) > validation.DNS952LabelMaxLength {
> > +		name = name[:validation.DNS952LabelMaxLength]
>
> We should only truncate a default name, not a user specified name (if you
> explicitly set one that is too long we should fail). Can you add that logic
> and a distinct test case for it?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/14728/files#r40997045>.
>